src/NzbDrone.Core/Music/TrackService.cs

```diff
@@ -81,60 +82,70 @@ public List<Track> GetTracksByAlbum(int albumId)
         public Track FindTrackByTitle(int artistId, int albumId, int mediumNumber, int trackNumber, string releaseTitle)
         {
             // TODO: can replace this search mechanism with something smarter/faster/better
-            var normalizedReleaseTitle = Parser.Parser.NormalizeTrackTitle(releaseTitle).Replace(".", " ");
+            var normalizedReleaseTitle = releaseTitle.NormalizeTrackTitle().Replace(".", " ");
             var tracks = _trackRepository.GetTracksByMedium(albumId, mediumNumber);
 
-            var matches = from track in tracks
-                //if we have a trackNumber use it
-                let trackNumCheck = (trackNumber == 0 || track.AbsoluteTrackNumber == trackNumber)
-                //if release title is longer than track title
-                let posReleaseTitle = normalizedReleaseTitle.IndexOf(Parser.Parser.NormalizeTrackTitle(track.Title), StringComparison.CurrentCultureIgnoreCase)
-                //if track title is longer than release title 
-                let posTrackTitle = Parser.Parser.NormalizeTrackTitle(track.Title).IndexOf(normalizedReleaseTitle, StringComparison.CurrentCultureIgnoreCase)
-                where track.Title.Length > 0 && trackNumCheck && (posReleaseTitle >= 0 || posTrackTitle >= 0)
-                orderby posReleaseTitle, posTrackTitle
-                select new
-                {
-                    NormalizedLength = Parser.Parser.NormalizeTrackTitle(track.Title).Length,
-                    Track = track
-                };
+            var matches = tracks.Where(t => (trackNumber == 0 || t.AbsoluteTrackNumber == trackNumber)
+                                       && t.Title.Length > 0
+                                       && (normalizedReleaseTitle.Contains(t.Title.NormalizeTrackTitle())
+                                           || t.Title.NormalizeTrackTitle().Contains(normalizedReleaseTitle)));
+
+            if (matches.Count() > 1)
+                return null;
 
-            return matches.OrderByDescending(e => e.NormalizedLength).FirstOrDefault()?.Track;
+            return matches.SingleOrDefault();
```

Comment:
Can simplify last 3 lines to `return matches.Count() > 1 ? null : matches.SingleOrDefault();`